From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aayush Gupta <theimpulson@e.email>
Date: Wed, 26 May 2021 10:17:03 +0200
Subject: [PATCH 01/32] Settings: res: Set proper title for system version

Signed-off-by: Aayush Gupta <theimpulson@e.email>
Change-Id: I3146e445e7b299086a566a66f8a13708699a91f7
Signed-off-by: althafvly <althafvly@gmail.com>
[a17: firmware_version.xml no longer lists the LineageOS version, so only
 my_device_info.xml is changed; the new catalyst LineageVersionDetailPreference.kt
 gets the same e_version title]

(cherry picked from commit 37cd5ab1dc8d8af7dbb2dacb779db0fd77c850cc)
---
 res/values/e_strings.xml                      | 20 +++++++++++++++++++
 res/xml/my_device_info.xml                    |  3 ++-
 .../LineageVersionDetailPreference.kt         |  2 +-
 3 files changed, 23 insertions(+), 2 deletions(-)
 create mode 100644 res/values/e_strings.xml

diff --git a/res/values/e_strings.xml b/res/values/e_strings.xml
new file mode 100644
index 0000000000..237a43c11b
--- /dev/null
+++ b/res/values/e_strings.xml
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~ Copyright Murena SAS 2023
+  ~ This program is free software: you can redistribute it and/or modify
+  ~ it under the terms of the GNU General Public License as published by
+  ~ the Free Software Foundation, either version 3 of the License, or
+  ~ (at your option) any later version.
+  ~
+  ~ This program is distributed in the hope that it will be useful,
+  ~ but WITHOUT ANY WARRANTY; without even the implied warranty of
+  ~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  ~ GNU General Public License for more details.
+  ~
+  ~ You should have received a copy of the GNU General Public License
+  ~  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+  -->
+<resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
+    <!-- /e/ version -->
+    <string name="e_version">/e/ OS version</string>
+</resources>
diff --git a/res/xml/my_device_info.xml b/res/xml/my_device_info.xml
index 9148a9d23f..f18a324ec2 100644
--- a/res/xml/my_device_info.xml
+++ b/res/xml/my_device_info.xml
@@ -149,8 +149,9 @@
         <Preference
             android:key="lineage_os_version"
             android:order="43"
-            android:title="@*lineageos.platform:string/lineage_version"
+            android:title="@string/e_version"
             android:summary="@string/summary_placeholder"
+            android:selectable="false"
             settings:enableCopying="true"
             settings:controller="com.android.settings.deviceinfo.firmwareversion.LineageVersionDetailPreferenceController"/>
     </PreferenceCategory>
diff --git a/src/com/android/settings/deviceinfo/firmwareversion/LineageVersionDetailPreference.kt b/src/com/android/settings/deviceinfo/firmwareversion/LineageVersionDetailPreference.kt
index 913207039f..365cca3b45 100644
--- a/src/com/android/settings/deviceinfo/firmwareversion/LineageVersionDetailPreference.kt
+++ b/src/com/android/settings/deviceinfo/firmwareversion/LineageVersionDetailPreference.kt
@@ -47,7 +47,7 @@ class LineageVersionDetailPreference :
         get() = R.string.os_firmware_version_purpose
 
     override val title: Int
-        get() = org.lineageos.platform.internal.R.string.lineage_version
+        get() = R.string.e_version
 
     override val indexable
         get() = false
